### PR TITLE
Frontend layout -- Basic API implementation

### PR DESCRIPTION
**Pull request by leftl**
More changes that were partially added from Thomas's UI work that was based on this branch.
We can now initialize a session with the backend API (only running on the same host), and fetch character data related to that session.
This is still a WIP--It is not currently populating the UI or pushing changes to the backend.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "frontend",
   "homepage": "https://ngutn24.github.io/RollBuddy",
+  "proxy": "http://localhost:4567",
```
Comment (Esauidk) [inline frontend/package.json:4]: Should this be backend's port number? 4567 is backend, 3030 is frontend

File: frontend/src/components/CharacterSheet.js [hunk 2]
```diff
@@ -1,29 +1,46 @@
...
-      // setCharacterID(id);
+      setCharacterID(id);
     } else {
-      localStorage.setItem("characterID", JSON.stringify(id));
-    }
-  }, []);
+      // TODO: Hardcoded URL, use github env variables in the future?
+      // NOTE: withCredentials will store the session cookie for later
+      axios
+        .get("http://localhost:4567/createw", { withCredentials: true })
```
Comment (Esauidk) [inline frontend/src/components/CharacterSheet.js:21]: typo, should be `/create`

**Review by Esauidk — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 1 above)
Comment (leftl) [reply frontend/package.json:4]: yes it's meant to proxy to the api endpoint!

File: frontend/src/components/CharacterSheet.js (hunk 2 above)
Comment (leftl) [reply frontend/src/components/CharacterSheet.js:21]: thanks, forgot to change after testing!

**Review by Esauidk — APPROVED** [review]
👍

**Pull request merged**